spec/jobs/daily_likes_digest_job_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'rails_helper'
+
+RSpec.describe DailyLikesDigestJob, type: :job do
+  it 'envia e-mail com resumo diário de curtidas' do
+    user = create(:user)
+    post_a = create(:post, user:)
+    post_b = create(:post, user:)
+    comment = create(:comment, user:)
+    3.times { create(:like, likeable: post_a) }
+    5.times { create(:like, likeable: post_b) }
+    2.times { create(:like, likeable: comment) }
+    mail = double('mail', deliver: true)
+    mailer = double('LikesMailer', notify_new_like: mail)
+    allow(LikesMailer).to receive(:with).and_return(mailer)
+    allow(mailer).to receive(:notify_like).and_return(mail)
+
+    DailyLikesDigestJob.perform_now(user:)
+
+    expect(mail).to have_received(:deliver).once
+  end
+
+  it 'não envia e-mail caso o usuário não tenha nenhuma curtida nas últimas 24 horas' do
+    user = create(:user)
+    travel_to(2.days.ago) do
+      post = create(:post, user:)
```

Comment:
Pessoal, fiquei confuso com esse teste. Criar likes há dois dias atrás e testar que nada foi enviado, já que nenhum like foi recebido no ultimo dia, talvez seja o mesmo que não criar nada e esperar que nenhum email foi recebido, não? Como por exemplo allow(Like).to receive(:where).with('created_at >= ?', 24.hours.ago).and_return([]) . Ví que tem um método para verificar os likes do post em um prazo, talvez dê p/ adaptar essa linha com ele.🍀

Reply:
A ideia desse teste é exatamente validar que existindo likes anteriores (que na teoria, foram notificados em outros dias) no banco de dados, o sistema não vai enviar um novo e-mail para o usuário por causa desses testes pré-existentes. Optamos por testar com o cenário de "não envia", porque isso já confirma que caso o usuário receba likes em um dia, os anteriores não entrarão na lógica do Mailer.

Eu não entendi a questão do mock com `allow(Like)`, fiquei confuso. A gente mockaria a curtida?

Reply:
Entendi, @eliseuramos93. Ta beleza, então. A linha de código foi um exemplo que tentei montar p/ testar a pesquisa de likes no ultimo dia. Mas desconsidera, por favor. Valeu!

Reply:
@caiquedv , tá tranquilo kkkk eu só realmente não entendi a lógica do usar o mock, queria entender pq talvez você esteja vendo algo que eu não tô, mas eu buguei pensando na questão de mockar o like kkkkk